api/src/permissions/utils/filter-policies-by-ip.ts

```diff
@@ -0,0 +1,19 @@
+import { ipInNetworks } from '../../utils/ip-in-networks.js';
+import type { AccessRow } from '../modules/process-ast/types.js';
+
+export function filterPoliciesByIp(policies: AccessRow[], ip: string | null | undefined) {
+	return policies.filter(({ policy }) => {
+		// Keep policies that don't have an ip address allow list configured
+		if (!policy.ip_access) {
```

Comment:
When all IPs are removed, the `ip_access` field is set to an empty array, resulting in the policy being filtered out.
Note: the typing for `policy.ip_access` is `string[] | null`.

```suggestion
		if (!policy.ip_access || policy.ip_access.length === 0) {
```

Reply:
Good catch!